rtil/src/threads/ue/rebo/livesplit.rs

```diff
@@ -97,4 +125,54 @@ impl Run {
     pub fn attempt_count() -> u32 {
         LIVESPLIT_STATE.lock().unwrap().timer.run().clone().attempt_count()
     }
+    pub fn save_splits(path: impl AsRef<Path>) -> Result<(), SplitsSaveError> {
+        let state = LIVESPLIT_STATE.lock().unwrap();
+        let path = path.as_ref();
+        let filename = path
+            .file_name().expect("Could not get filename")
+            .to_str().expect("Could not convert filename to &str")
+            .to_owned();
+        let filename = sanitize_filename::sanitize(filename);
+        let valid_splits = VALID_SPLITS.lock().unwrap();
+        let path = match path.is_relative() || valid_splits.contains(&path.to_path_buf()) {
+            true => LiveSplit::splits_path().join(&filename),
+            false => {
+                if valid_splits.contains(&path.to_path_buf()) {
+                    path.to_path_buf()
+                } else {
+                    return Err(SplitsSaveError::DisallowedFilePath(filename, "Disallowed file path".to_owned()));
+                }
+            },
+        };
+        let file = match File::create(path) {
+            Ok(file) => file,
+            Err(e) => return Err(SplitsSaveError::CreationFailed(filename, e.to_string())),
+        };
```

Comment:
∀ The pattern of using the inner `Ok`-value and returning a mapped `Err`-value can be expressed more easily using
```rust
let file = Filed:create(path).map_err(|e| SplitsSaveError::CreationFailed(filename, e.to_string()))?;
```